chunk.go

```diff
@@ -0,0 +1,41 @@
+package typhon
+
+import (
+	"io"
+	"net/http"
+)
+
+func copyChunked(dst io.Writer, src io.Reader) (written int64, err error) {
+	flusher, flusherOk := dst.(http.Flusher)
+	if !flusherOk {
+		return io.Copy(dst, src)
+	}
+
+	// This is taken and lightly adapted from the source of io.Copy
+	buf := make([]byte, 32*1024)
```

Comment:
This may be a very ignorant question, but I'd like to know. Why is this 32*1024?

Reply:
I'm honestly not sure. I copied this code directly from the `io` package. I just went and looked up the history and I can see that this code dates [all the way back to 2008](https://github.com/golang/go/commit/79d94d504f8f3e82e994a4f63d37f56cebc6e7cc). Sadly the commit message doesn't provide any information as to _why_ this number was chosen, but I suspect it's a somewhat arbitrary choice that also is a multiple of the default page size in Linux and most other operating systems.